<?xml version="1.0" encoding="utf-8"?>

<launch>

  <!-- Load Parameter -->
  
  <param name="use_cpu"           value="true" />

  <!-- Start yolov5 and ros wrapper -->
  <node pkg="yolov8_ros" type="yolo_v8.py" name="yolov8_ros" output="screen" >
    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/yolov8n_openvino_model"/> -->
    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/yolov8n.onnx"/> -->
    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/yolov8n.pt"/> -->

    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/best_v8n.pt"/> -->
    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/best_v8n_sim.pt"/> -->
    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/best_v8n_sim_pretrain.pt"/> -->

    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/best_v8n_sim_128.pt"/> -->
    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/best_v8n_sim_pretrain_128.pt"/> -->

    <!-- <param name="weight_path"       value="$(find yolov8_ros)/weights/best_v8n_sim_batch16.pt"/> -->
    <param name="weight_path"       value="$(find yolov8_ros)/weights/best_v8n_sim_pretrain_batch16.pt"/>

    <param name="image_topic"       value="/usb_cam/image_rect" />
    <param name="pub_topic"         value="/yolov8/BoundingBoxes" />
    <param name="camera_frame"       value="camera_color_frame"/>
    <param name="visualize"       value="false"/>
    <param name="conf"              value="0.3" />
    <param name="imgsz"              value="320" />
  </node>

</launch>
